fitbenchmarking/fitbenchmark_one_problem.py

```diff
@@ -28,59 +37,106 @@ def fitbm_one_prob(user_input, problem):
                 containing the fit information
     """
 
-    previous_name, count = None, 0
     results_fit_problem = []
-    data_struct, cost_function, function_definitions = \
-        prereq.prepare_software_prerequisites(user_input.software, problem,
-                                              user_input.use_errors)
 
-    for function in function_definitions:
+    software = user_input.software.lower()
+
+    controllers = {'mantid': MantidController,
+                   'sasview': SasviewController,
+                   'scipy': ScipyController}
+
+    if software in controllers:
+        controller = controllers[software](problem, user_input.use_errors)
+    else:
+        raise NotImplementedError('The chosen software is not implemented yet: {}'.format(user_input.software))
+
+    problem.data_x = controller.data_x
```

Comment:
In line 49 problem is parsed to the initialisation of a controller, but then the controller is used to update the problem object. This looks odd. As a minimum please explain why this is done